code/lib/cli/scripts/update-core-portal.ts

```diff
@@ -0,0 +1,103 @@
+import { join } from 'node:path';
+import { sortPackageJson } from '../../../../scripts/node_modules/sort-package-json';
+
+import { readJSON, writeFile, ensureFile } from 'fs-extra';
+import dedent from 'ts-dedent';
+
+const write = async (location: string, data: string) => {
+  await ensureFile(location);
+  return writeFile(location, data);
+};
+
+const mapCoreExportToSelf = (map: Record<string, string>) => {
+  return Object.entries(map).reduce<Record<string, string>>((acc, [key, input]) => {
+    const value = input.replace('./dist/', './core/');
+    acc[key] = value;
+
+    return acc;
+  }, {});
+};
+
+const generateMapperContent = (input: string) => {
+  const value = input
+    .replace('./core/', '')
+    .replace('/index', '')
+    .replace('.cjs', '')
+    .replace('.d.ts', '')
+    .replace('.mjs', '')
+    .replace('.js', '');
+  if (input.endsWith('.js')) {
+    return `export * from '@storybook/core/${value}';\n`;
+  }
+  if (input.endsWith('.cjs')) {
+    return `module.exports = require('@storybook/core/${value}');\n`;
+  }
+  if (input.endsWith('.d.ts')) {
+    return dedent`
+      export * from '@storybook/core/${value}';
+      export type * from '@storybook/core/${value}';\n
+    `;
+  }
+  // eslint-disable-next-line local-rules/no-uncategorized-errors
+  throw new Error(`Unexpected input: ${input}`);
+};
+
+async function run() {
```

Comment:
Can you please leave a comment which describes roughly what this script is supposed to do?

Reply:
See:
https://github.com/storybookjs/storybook/blob/df82a5894dacb399405451a4e4be0714900f791d/code/lib/cli/scripts/update-core-portal.ts#L7-L17